anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -483,201 +482,240 @@ def image(dbsession, request_inputs, bodycontent=None):
+                "input dockerfile data must be base64 encoded - exception on decode: "
+                + str(err)
+            )
+    return dockerfile, dockerfile_mode
+
+
+def resolve_image_info_overrides_and_input_string(params, image_info, input_string):
+    image_info_overrides = {}
+
+    input_tag = params.get("tag", None)
+    input_digest = params.get("digest", None)
+    if input_tag and input_digest:
+        input_fulldigest = "{}/{}@{}".format(
+            image_info["registry"], image_info["repo"], input_digest
+        )
+        image_info_overrides["fulltag"] = input_tag
+        image_info_overrides["tag"] = image_info["tag"]
+        if params.get("created_at", None):
+            image_info_overrides["created_at_override"] = params.get("created_at")
+        input_string = input_fulldigest
+    return image_info_overrides, input_string
+
+
+def resolve_final_image_info(
+    account_id, input_string, registry_creds, dbsession, image_info_overrides
+):
+    try:
+        image_info = anchore_engine.common.images.get_image_info(
+            account_id,
+            "docker",
+            input_string,
+            registry_lookup=True,
+            registry_creds=registry_creds,
         )
-        httpcode = err.__response_code__
     except Exception as err:
-        logger.exception("Error processing image request")
-        return_object = anchore_engine.common.helpers.make_response_error(
-            err, in_httpcode=httpcode
+        fail_event = anchore_engine.subsys.events.ImageRegistryLookupFailed(
+            user_id=account_id,
+            image_pull_string=input_string,
+            data=err.__dict__,
+        )
+        try:
+            add_event(fail_event, dbsession)
+        except Exception:
+            logger.warn("Ignoring error creating image registry lookup event")
+        raise err
+
+    if image_info_overrides:
+        image_info.update(image_info_overrides)
+
+    return image_info
+
+
+def validate_image_size(image_info):
+    if not is_image_valid_size(image_info):
+        localconfig = anchore_engine.configuration.localconfig.get_config()
+        raise BadRequest(
+            "Image size is too large based on max size specified in the configuration",
+            detail={
+                "requested_image_compressed_size_mb": anchore_utils.bytes_to_mb(
+                    image_info["compressed_size"], round_to=2
+                ),
+                "max_compressed_image_size_mb": localconfig.get(
+                    "max_compressed_image_size_mb"
+                ),
+            },
         )
 
-    return return_object, httpcode
+
+def get_manifest(image_info):
+    try:
+        if "manifest" in image_info:
+            manifest = json.dumps(image_info["manifest"])
+        else:
+            raise Exception("no manifest from get_image_info")
+    except Exception as err:
+        raise Exception("could not fetch/parse manifest - exception: " + str(err))
```

Comment:
This is similar to the other exception handling, but causes a problem: the original exception is being caught and only added in its string representation to a new error that adds a message, obfuscating the originating exception. This will cause debugging to be really hard